urllib3/exceptions.py

```diff
@@ -196,6 +196,14 @@ class ResponseNotChunked(ProtocolError, ValueError):
     pass
 
 
+class BodyNotHttplibCompatible(AssertionError, ValueError):
```

Comment:
Well, the problem (as you can see from this PR after my requested changes) is that not every exception was inheriting from that to begin with. The pattern that @frankier chose to start with was already present (and kind of weird, at that).

We can't add HTTPError to those exception classes otherwise we'll marginally break people who have exceptions structured like


```py
try:
    # ...
except urllib3.exceptions.HTTPError:
    # ...
except urllib3.exceptions.ProxySchemeUnknown:
    # ...
```

So I do think we might want to restructure (although maybe right now is not the best time).